### DISCUSSION TITLE
With snack introduction, how do we handle keymaps?

### DISCUSSION THREAD

**Discussion by danjessen** [LazyVim/LazyVim#4766] [Q&A]
I guess the title says it all. Snack has some keymaps added directly to the keymaps.lua. They don't test if the the module is enabled.

Are we supposed to remap them or should keymaps uphold the config? Thirdly should they live in keymaps or in the plugin definition? Right now it's both the config settings and keymaps are all over the place:

- https://github.com/LazyVim/LazyVim/blob/fbf881f80b8f0fb2d9e4bc69d04f66323b25c4b9/lua/lazyvim/config/keymaps.lua#L40
- https://github.com/LazyVim/LazyVim/blob/220148f47f175229232eee31982577fbdd637663/lua/lazyvim/plugins/init.lua#L36
- https://github.com/LazyVim/LazyVim/commit/220148f47f175229232eee31982577fbdd637663#diff-59200cd613b2ec9502ad86d5b68f569fd74826c80b51c8fb6353b913c73a652cL34

**Comment by dpetka2001** [comment]
The ones in `keymaps.lua` are global mappings. The ones in the `snacks.nvim` plugin spec only apply to the specific snack (i.e only for the `Snacks.terminal` in the case that you linked in `/lua/lazyvim/plugins/init.lua`).

**Comment by danjessen** [comment]
@dpetka2001 I get that. Is it expected that we remove / del global keymaps. Or will they be able to look at Snack.config. Much like the last link, to the commit?

**Reply by dpetka2001** [reply]
You have to change/delete(disable) mappings in the place where they're defined.  So, if you want to disable `<leader>bd` you do that in `keymaps.lua` and if you want to change the `Snacks.terminal` mappings you do that in the plugin spec. If i don't understand you correctly, please clarify what you're trying to achieve.

**Reply by danjessen** [reply]
It is what I am asking. My follow up question is. Would it make sense for LazyVim to respect the Snacks.config, in regards to keymaps, since it already does so in some places?
